docs/topics/faq.rst

```diff
@@ -0,0 +1,95 @@
+
+**Q**:. How to choose between synchronous(DoQuery) and asynchronous(SubmitQuery/FetchResult) mode?
+
+   Synchronous mode is used for small data volumes, where you directly obtain the results through the response after submitting the query. Asynchronous mode is used for large data volumes, where the execution time is long, to avoid request timeouts. After submitting the query, you need to repeatedly call fetch result to check the results.
+
+   **Note:** If the number of results returned by the query exceeds 10w, please choose the asynchronous mode to avoid potential issues.
+
+**Q**: What data sources does the engine support?
+
+   - SCQL directly supports the following data sources:
+
+   1. CSV (including local files, OSS, Minio)
+   2. MySQL and databases compatible with the MySQL protocol
+   3. Postgres
+   
+   - SCQL can be extended to support the following data sources:
+
+   4. On Kuscia, additional support for ODPS. **NOTE:** When using Kuscia, users can register data source information (such as CSV file locations, database connection string of MySQL and Postgres) in Kuscia DomainData. SCQL can then access this information through Kuscia Datamesh and process it accordingly.
+   5. SCQL supports the Arrow SQL client, and users can implement their own data sources by providing an Arrow SQL server
+
+Errors Occurred During Execution
+--------------------------------
+
+**Q**: The engine reported a "Get data timeout" error during execution.
+
+   It is necessary to troubleshoot based on the specific situation, whether the request was intercepted by the gateway, or if there was an error in the execution of the engine on the other side. It could also be due to poor network conditions. If the issue is caused by poor network conditions, you can alleviate this error by modifying the relevant network configuration. Please refer to the configuration documentation for detailed settings :doc:`/reference/p2p-deploy-config` :doc:`/reference/centralized-deploy-config`.
+
+**Q**: SCQL results from executing group by related syntax are incomplete or do not match the MySQL results?
+
+   SCQL, to protect data privacy and prevent the malicious theft of data within groups, hides groups where the number of data items within a group is less than the GroupByThreshold. For specific details, please refer to the security_compromise.group_by_threshold configuration option in the documentation :doc:`/reference/p2p-deploy-config` (by default, groups with fewer than 4 data items are not displayed). Setting this value to 1 will disable the group filtering operation.
+
+**Q**: There are precision errors in the numerical calculations.
+
+   When SCQL enters secure MPC protocol, it needs to encode the data into Ring64 or Ring128 and then perform the secure computation. Numerical inaccuracies can occur during both the encoding and the secure computation processes, and this is unavoidable.
```

Comment:
Nit: add a reference link to 
1. https://www.secretflow.org.cn/zh-CN/docs/spu/0.9.3b0/getting_started/faq#i-have-met-huge-inconsistency-between-spu-result-and-plaintext-jax-result
2. https://www.secretflow.org.cn/en/docs/spu/0.9.3b0/development/fxp

Reply:
will add references and add Chinese doc in next pr